### PR TITLE
[ Milestone 2 ] Implementation

### PR DESCRIPTION
**Pull request by CakeInTech**
## Implement adds and removes to the todo-list.

- Removed all hardcoded items from the tasks array.
- Created a new JavaScript file for the new functionality.
- Implemented a function for adding a new task (add a new element to the array).
- Implemented a function for deleting a task (remove an element from the array).
- Implemented a function for editing task descriptions.
- All changes to the To Do List should be saved in local storage.

## Request review changes

-  Add Readme.MD file.
- I accidentally Merged my previous pr from #2

### CODE REVIEW

**Review by FrederickMih — COMMENTED** [review]
Hi @MohamedCK , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

**You have submitted a project, that doesn't contain a README file.**
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by CakeInTech** [comment]
@FrederickMih  thank you for the reminder i should add it

File: src/modules/clearcm.js [hunk 1]
```diff
@@ -22,7 +22,4 @@ export const clearAllTask = () => {
   });
   task.push(...newArr);
   localStorage.setItem('task', JSON.stringify(task));
```
Comment (DeliceLydia) [inline src/modules/clearcm.js:24]: - [x] I am not able to edit a to do please work on the `edit` method
- [x] Make sure the property `index` should be set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5).
- [x] When you delete a task make sure that it update all remaining items' indexes

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @MohamedCK, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Highlights for your project:
- Awesome README
- Followed git workflow


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] I highly suggest that you commit any meaningful change in your project it shows that you are dedicated to it


### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by uwadonat — APPROVED** [review]
## Hi @MohamedCK,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
- Professional readme file ✔️
- No linter error ✔️
- Descriptive pull request message ✔️
good-job-awesome
# Status:  Approved  ✔️✔️✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**